services/staging/rs_server_staging/processors.py

```diff
@@ -677,14 +680,33 @@ def dask_cluster_connect(self):
                 )
                 clusters = gateway.list_clusters()
                 self.logger.debug(f"The list of clusters: {clusters}")
-                self.cluster = gateway.connect(clusters[0].name)
+
+                # Get the identifier of the cluster whose name is equal to the cluster_name variable
+                # Protection for the case when this cluster does not exit
+                cluster_id = [
+                    cluster.name
+                    for cluster in gateway.list_clusters()
+                    if (
+                        isinstance(cluster.options, dict)
+                        and "cluster_name" in cluster.options
+                        and cluster.options["cluster_name"] == cluster_name
+                    )
+                ][0]
+                self.cluster = gateway.connect(cluster_id)
```

Comment:
what if cluster_id is still None?

Reply:
```python
cluster_id = next(
    (
        cluster.name
        for cluster in gateway.list_clusters()
        if isinstance(cluster.options, dict)
        and cluster.options.get("cluster_name") == cluster_name
    ),
    None,  # Default value if no matching cluster is found
)

if cluster_id is None:
    raise ValueError(f"No cluster found with name: {cluster_name}")
```

Reply:
It's a bit unclear, since you raise keyerror and indexerror at the end.

Maybe remove the whole try-except block, and raise only if cluster_id is none

Reply:
I followed the suggestion